llvm/test/CodeGen/AArch64/complex-deinterleaving-cdot.ll

```diff
@@ -0,0 +1,170 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
```

Comment:
Could you add a more focused test without the loop?

Reply:
I've removed as much of the loop as I can, however we need the IR to at least resemble a loop in order for the reduction to be valid and recognised.